pipeline/rds_to_s3.py

```diff
@@ -0,0 +1,125 @@
+"""
+Contains code to retrieve watering/recording data older than 24hrs from the database, append it to
+the relevant files in s3 (automatically creates if none exist), and delete from the db.
+"""
+
+from os import environ
+from datetime import datetime, timedelta
+import pandas as pd
+from dotenv import load_dotenv
+import sqlalchemy as db
+from sqlalchemy.engine.base import Connection
+from boto3 import client
+
+
+load_dotenv()
+STR_DAY_AGO = (datetime.now() - timedelta(hours=24)).strftime("%Y-%m-%d %H:%M:%S")
+TODAY = datetime.today()
+
+
+def get_database_engine():
+    """Returns the database engine."""
+    try:
+        engine = db.create_engine(
+            f"mssql+pymssql://{environ['DB_USER']}:{environ['DB_PASSWORD']}@{environ['DB_HOST']}:{environ['DB_PORT']}/{environ['DB_NAME']}?charset=utf8"
+        )
+        return engine
+    except Exception as e:
+        print(f"Error creating database engine: {e}")
+        raise e
+
+
+def get_old_records(table_name: str, db_engine: db.Engine, connection: Connection, metadata,
+                    date_cutoff: str = STR_DAY_AGO):
```

Comment:
Nitpick - but PEP8 says to not put spaces around '=' in arguments...

[Don’t use spaces around the = sign when used to indicate a keyword argument, or when used to indicate a default value for an unannotated function parameter:](https://peps.python.org/pep-0008/#other-recommendations)

```
# Correct:
def complex(real, imag=0.0):
    return magic(r=real, i=imag)
```

```
# Wrong:
def complex(real, imag = 0.0):
    return magic(r = real, i = imag)
```

Reply:
That's only the case if there's no type hints; from directly below the bit you've attached:

<img width="775" alt="Screenshot 2023-12-19 at 13 55 11" src="https://github.com/caitlinturnidge/lmnh-plant-pipeline/assets/143012198/2197b0c8-a829-44f0-abc8-271ae340f0c6">